File: 2149_Rearrange_Array_Elements_by_Sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        queue<int> pos_q;
        queue<int> neg_q;
        int i = 0;
        while(i < nums.size())
        {
            if(nums[i] >= 0) pos_q.push(nums[i]);
            else neg_q.push(nums[i]);
            i++;
        }
        i = 0;
        while(!pos_q.empty())
        {
            nums[i] = pos_q.front();
            nums[i + 1] = neg_q.front();
            pos_q.pop();
            neg_q.pop();
            i += 2;
        }
        return nums;
    }
};

File: 2149_Rearrange_Array_Elements_by_Sign.js
var rearrangeArray = function(nums) {
    let pos = []
    let neg = []
    let i = 0
    while(i < nums.length)
    {
        if(nums[i] >= 0) pos.push(nums[i])
        else neg.push(nums[i])
        i++
    }
    let ans = []
    i = 0
    while(i < (nums.length / 2))
    {
        ans.push(pos[i])
        ans.push(neg[i])
        i++
    }
    return ans
};
